code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1610,87 +1610,240 @@ GLOBAL_LIST_EMPTY(roundstart_races)
-/datum/species/proc/handle_body_temperature(mob/living/carbon/human/H)
+		// Are we sweaty, check against unmodified body temp.
+		if(humi.get_body_temp_normal(apply_change=FALSE) + 10 < humi.coretemperature) // with small buffer as temp can bounce a bit
+			// we are overheating and sweaty insulation is not as good reducing thermal protection
+			area_skin_change = (1 - (thermal_protection * 0.7)) * area_skin_change
+		else
+			area_skin_change = (1 - thermal_protection) * area_skin_change
+
+		humi.adjust_bodytemperature(area_skin_change)
+
+	// Core to skin temp transfer, when not on fire
+	if(!humi.on_fire)
+		// Get the changes to the skin from the core temp
+		var/core_skin_diff = humi.coretemperature - humi.bodytemperature
+		var/core_skin_change = (1 + thermal_protection) * get_temp_change_amount(core_skin_diff, 0.08)
+
+		// We do not want to over shoot after using protection
+		if(core_skin_diff > 0)
+			core_skin_change = min(core_skin_change, core_skin_diff)
+		else
+			core_skin_change = max(core_skin_change, core_skin_diff)
+
+		humi.adjust_bodytemperature(core_skin_change)
+
+
+/**
+ * Used to get the amount of change between two body temperatures
+ *
+ * When passed the difference between two temperatures returns the amount of change to temperature to apply.
+ * The change rate should be kept at a low value tween 0.16 and 0.02 for optimal results.
+ * vars:
+ * * temp_diff (required) The differance between two temperatures
+ * * change_rate (optional)(Default: 0.06) The rate of range multiplyer
+ */
+/datum/species/proc/get_temp_change_amount(var/temp_diff, var/change_rate = 0.06)
+	if(temp_diff < 0)
+		return (log((temp_diff * -1) * change_rate + 1) * BODYTEMP_AUTORECOVERY_DIVISOR) * -1
+	return log(temp_diff * change_rate + 1) * BODYTEMP_AUTORECOVERY_DIVISOR
+
+/**
+ * Used to set alerts and debuffs based on body temperature
+ * vars:
+ * * humi (required) The mob we will targeting
+ */
+/datum/species/proc/body_temperature_alerts(mob/living/carbon/human/humi)
 	// Body temperature is too hot, and we do not have resist traits
-	if(H.bodytemperature > bodytemp_heat_damage_limit && !HAS_TRAIT(H, TRAIT_RESISTHEAT))
+	if(humi.bodytemperature > bodytemp_heat_damage_limit && !HAS_TRAIT(humi, TRAIT_RESISTHEAT))
 		// Clear cold mood and apply hot mood
-		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "cold")
-		SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "hot", /datum/mood_event/hot)
+		SEND_SIGNAL(humi, COMSIG_CLEAR_MOOD_EVENT, "cold")
+		SEND_SIGNAL(humi, COMSIG_ADD_MOOD_EVENT, "hot", /datum/mood_event/hot)
 
 		//Remove any slowdown from the cold.
-		H.remove_movespeed_modifier(/datum/movespeed_modifier/cold)
+		humi.remove_movespeed_modifier(/datum/movespeed_modifier/cold)
+		// display alerts based on how hot it is
+		switch(humi.bodytemperature)
+			if(0 to 461)
+				humi.throw_alert("temp", /obj/screen/alert/hot, 1)
+			if(460 to 700)
+				humi.throw_alert("temp", /obj/screen/alert/hot, 2)
```

Comment:
Should this really be hardcoded? And what about different species? Should stable body temp be taken into account here?

Reply:
The bodytemp_heat_damage_limit for the species is checked before this, and switch's do not accept calculations.
This is meant to spread the alerts as temp increases it starting at 0 is just simplicity as we cant get here without being above the bodytemp_heat_damage_limit.

Reply:
Would it accept a reference?
Like, if(species.minsafetemperature to species.maxsafetemperature) or something?

Reply:
That does work, but we only have one data point, when to start taking damage.
We do not have a second and third degree value per species, and i do not feel it adds anything to have 4 more vars on a species just to set when the alert increases.